sktime/experiments/analysis.py

```diff
@@ -0,0 +1,374 @@
+
+import pandas as pd
+import numpy as np
+import itertools
+
+from scipy import stats
+from scipy.stats import ttest_ind
+from scipy.stats import ranksums
+
+from sklearn.metrics import accuracy_score, mean_squared_error
+import scikit_posthocs as sp
+
+from sktime.experiments.losses import Losses
+
+import matplotlib.pyplot as plt
+
+class AnalyseResults(object):
+    """
+    Analyze results of machine learning experiments.
+
+    Parameters
+    ----------
+    result: sktime result object
+        class for storing the results
+    """
+
+    def __init__(self, 
+                 results):
+
+        self._results_list = results.load()
+    
+    def prediction_errors(self, metric):
+        """
+        Calculates the average prediction error per estimator as well as the prediction error achieved by each estimator on individual datasets.
+
+        Parameters
+        -----------
+        metric: `sktime.analyse_results.scores`
+            Error function 
+        Returns
+        --------
+        pickle of pandas DataFrame
+            ``estimator_avg_error`` represents the average error and standard deviation achieved by each estimator. ``estimator_avg_error_per_dataset`` represents the average error and standard deviation achieved by each estimator on each dataset.
+        """
+        #load all predictions
+        losses = Losses(metric)
+        for res in self._results_list:
```

Comment:
```suggestion
        for res in self._results_list.load():
```
The `load` method would have to be an iterator then.